### ISSUE TITLE
Product Collection: new collection Related Products

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#49335]
**Background:**

There's Related Products block that is a Query Loop variation. It has limitations and don't come with the PC features.

**Goal:**

The goal of this issue is to prepare a PC collection called Related Products. This is also to explore topic of product-context collections. The outcome of this issue should include some guidelines and potential technical gaps preventing us from implementing Upsells and Cross-Sells collections.

**Collection Requirements:**

- It follows the layout of the current Related Products block
- Initially, it should use the same/similar logic of displaying products
- It works in the scope of:
    - Single Product block
    - Single Product template
- For now mark is experimental, and shouldn't be exposed in the inserter until the product context chooser is available (in scope of a separate issue).

**Comment by ObliviousHarmony** [comment]
I've spent some time diving into the existing `woocommerce/related-products` block and I _think_ this issue is blocked but will be really trivial once https://github.com/woocommerce/woocommerce/issues/44877 has been merged. 

The entire block mostly boils down to [this function that gets the related products](https://github.com/woocommerce/woocommerce/blob/cd638721f9e133b2cf2f15ebb22157441a14127b/plugins/woocommerce/src/Blocks/BlockTypes/RelatedProducts.php#L158-L174) via `wc_get_related_products()` and adds the IDs as a filter in the query. The main point of difficulty in implementing the block is the need for a product context, which https://github.com/woocommerce/woocommerce/issues/44877 is going to add. Once that has happened we can bring over the query modification code, adjust it to use the product context, and be on our way.

**Comment by imanish003** [comment]
@ObliviousHarmony Thanks a lot for the update. It definitely helps to know that we can use existing logic for fetching related products 🙌🏻

IMO, It would be great if we could create a draft PR assuming that the product ID is available for us to use. Maybe we can use random product ID for now. Basically, there are 2 cases:

1. **From the current product**: In this case, I believe we can use `productCollectionLocation` from context on Frontend, which is added [here](https://github.com/woocommerce/woocommerce/blob/f32fcaf23e27cf57f2f88b568416493e475106e5/plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php#L120-L141). This should provide the product ID on frontend when product ID is available at template level (ex. Single Product template) or provided by Single Product block (When collection is inside this block). 

2. **From a specific product**: This is where the context picker I am working on come into the play. Once [#50164](https://github.com/woocommerce/woocommerce/pull/50164) is merged, there will be product ID in `productReference` field of `query` attribute of the block. I believe for draft PR, we may use a dummy product ID. Once PR is merged, I believe change will be small to use `productReference` instead. 

This draft PR will also be a good opportunity to provide any early input or feedback. What do you think? 🙂
